matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/create/CreateLocalRoomTask.kt

```diff
@@ -0,0 +1,267 @@
+        try {
+            awaitNotEmptyResult(realmConfiguration, TimeUnit.MINUTES.toMillis(1L)) { realm ->
+                realm.where(RoomSummaryEntity::class.java)
+                        .equalTo(RoomSummaryEntityFields.ROOM_ID, roomId)
+                        .equalTo(RoomSummaryEntityFields.MEMBERSHIP_STR, Membership.JOIN.name)
+            }
+        } catch (exception: TimeoutCancellationException) {
+            throw CreateRoomFailure.CreatedWithTimeout(roomId)
+        }
+
+        return roomId
+    }
+
+    /**
+     * Create a local room entity from the given room creation params.
+     * This will also generate and store in database the chunk and the events related to the room params in order to retrieve and display the local room.
+     */
+    private suspend fun createLocalRoomEntity(realm: Realm, roomId: String, createRoomBody: CreateRoomBody) {
+        RoomEntity.getOrCreate(realm, roomId).apply {
+            membership = Membership.JOIN
+            chunks.add(createLocalRoomChunk(realm, roomId, createRoomBody))
+        }
+    }
+
+    private fun createLocalRoomSummaryEntity(realm: Realm, roomId: String, createRoomBody: CreateRoomBody) {
+        val otherUserId = createRoomBody.getDirectUserId()
+        if (otherUserId != null) {
+            RoomSummaryEntity.getOrCreate(realm, roomId).apply {
+                isDirect = true
+                directUserId = otherUserId
+            }
+        }
+        roomSummaryUpdater.update(
+                realm = realm,
+                roomId = roomId,
+                membership = Membership.JOIN,
+                roomSummary = RoomSyncSummary(
+                        heroes = createRoomBody.invitedUserIds.orEmpty().take(5),
+                        joinedMembersCount = 1,
+                        invitedMembersCount = createRoomBody.invitedUserIds?.size ?: 0
+                ),
+                updateMembers = !createRoomBody.invitedUserIds.isNullOrEmpty()
+        )
+    }
+
+    /**
+     * Create a single chunk containing the necessary events to display the local room.
+     *
+     * @param realm the current instance of realm
+     * @param roomId the id of the local room
+     * @param createRoomBody the room creation params
+     *
+     * @return a chunk entity
+     */
+    private suspend fun createLocalRoomChunk(realm: Realm, roomId: String, createRoomBody: CreateRoomBody): ChunkEntity {
+        val chunkEntity = realm.createObject<ChunkEntity>().apply {
+            isLastBackward = true
+            isLastForward = true
+        }
+
+        val eventList = createLocalRoomEvents(createRoomBody)
+        val eventIds = ArrayList<String>(eventList.size)
+        val roomMemberContentsByUser = HashMap<String, RoomMemberContent?>()
+
+        for (event in eventList) {
+            if (event.eventId == null || event.senderId == null || event.type == null) {
+                continue
+            }
+
+            eventIds.add(event.eventId)
```

Comment:
it looks like these `eventIds` aren't used? :eyes:

Reply:
you're right, it's not used anymore 👍🏻